README.md

```diff
@@ -208,181 +181,26 @@ The has_ancestry method supports the following options:
-
-    node.ancestors(:from_depth => -6, :to_depth => -4)
-    node.path.from_depth(3).to_depth(4)
-    node.descendants(:from_depth => 2, :to_depth => 4)
-    node.subtree.from_depth(10).to_depth(12)
-
-Please note that depth constraints cannot be passed to ancestor_ids and
-path_ids. The reason for this is that both these relations can be fetched
-directly from the ancestry column without performing a database query. It
-would require an entirely different method of applying the depth constraints
-which isn't worth the effort of implementing. You can use
-ancestors(depth_options).map(&:id) or ancestor_ids.slice(min_depth..max_depth)
-instead.
+    :touch                 Instruct Ancestry to touch the ancestors of a node when it changes, in order
+                           to invalidate nested key-based caches (default: false)
 
 # STI support
 
-Ancestry works fine with STI. Just create a STI inheritance hierarchy and
+Ancestry works fine with STI: just create a STI inheritance hierarchy and
 build an Ancestry tree from the different classes/models. All Ancestry
 relations that were described above will return nodes of any model type. If
-you do only want nodes of a specific subclass you'll have to add a condition
+you do only want nodes of a specific subclass, you'll have to add a condition
 on type for that.
 
-# Arrangement
-
-Ancestry can arrange an entire subtree into nested hashes for easy navigation
-after retrieval from the database.  `TreeNode.arrange` could for example return:
-
-```ruby
-{
-  #<TreeNode id: 100018, name: "Stinky", ancestry: nil> => {
-    #<TreeNode id: 100019, name: "Crunchy", ancestry: "100018"> => {
-      #<TreeNode id: 100020, name: "Squeeky", ancestry: "100018/100019"> => {}
-    },
-    #<TreeNode id: 100021, name: "Squishy", ancestry: "100018"> => {}
-  }
-}
-```
-
-The `arrange` method also works on a scoped class, for example:
-
-```ruby
-TreeNode.find_by_name('Crunchy').subtree.arrange
-```
-
-The `arrange` method takes `ActiveRecord` find options. If you want your hashes to
-be ordered, you should pass the order to the `arrange` method instead of to the
-scope. example:
-
-```ruby
-TreeNode.find_by_name('Crunchy').subtree.arrange(:order => :name)
-```
-
-To get the arranged nodes as a nested array of hashes for serialization:
-
-`TreeNode.arrange_serializable`
-
-```ruby
-[
-  {
-    "ancestry" => nil, "id" => 1, "children" => [
-      { "ancestry" => "1", "id" => 2, "children" => [] }
-    ]
-  }
-]
-```
-
-You can also supply your own serialization logic using blocks:
-
-For example, using `ActiveModel` Serializers:
-
-```ruby
-TreeNode.arrange_serializable do |parent, children|
-  MySerializer.new(parent, children: children)
-end
-```
-
-Or plain hashes:
-
-```ruby
-TreeNode.arrange_serializable do |parent, children|
-  {
-     my_id: parent.id,
-     my_children: children
-  }
-end
-```
-
-The result of `arrange_serializable` can easily be serialized to json with
```

Comment:
lets reduce these comments to what you think is relevant. but maybe we could keep here?